### PR TITLE
Fix an issue where clusterClient's temporary kubeconfig files could persist after the application terminates. Add comments for the clusterclient New(...) methods.

### PR DESCRIPTION
**Pull request by spew**
**What this PR does / why we need it**:
As pointed out by @roberthbailey on PR https://github.com/kubernetes-sigs/cluster-api/pull/384 there are opportunities for clusterclient to persist temporary files even after the application terminates. Namely in error scenarios while creating the clusterclient. This PR addresses those issues. 

**Special notes for your reviewer**:

**Release note**:
```release-note
NONE
```

@kubernetes/kube-deploy-reviewers

### CODE REVIEW

**Comment by roberthbailey** [comment]
lgtm (please fix the nit though)

**Comment by spew** [comment]
Hi @roberthbailey I don't see any comment on what to change.

File: clusterctl/clusterdeployer/clusterclient.go [hunk 1]
```diff
@@ -238,10 +243,22 @@ func createTempFile(contents string) (string, error) {
...
 	_, err = f.WriteString(contents)
 	if err != nil {
 		return "", err
 	}
 	return f.Name(), nil
 }
+
+func ifErrRemove(pErr *error, path string) {
+	if *pErr != nil {
+		err := os.Remove(path)
```
[the remark this answers is not in the document]
Comment (spew) [reply clusterctl/clusterdeployer/clusterclient.go]: Changed.

**Pull request merged**